File: OnlineCabService-Service/src/main/java/project/onlinecabservice/service/model/VehicleType.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package project.onlinecabservice.service.model;

/**
 *
 * @author Sumaiya
 */
public class VehicleType {
    private int TypeID;
    private String TypeName;
    private int vehicleCapacity;
    private int pricePerKmInLKR;

    public VehicleType(int TypeID) {
        this.TypeID = TypeID;
    }

    public VehicleType(int TypeID, String TypeName, int vehicleCapacity, int pricePerKmInLKR) {
        this.TypeID = TypeID;
        this.TypeName = TypeName;
        this.vehicleCapacity = vehicleCapacity;
        this.pricePerKmInLKR = pricePerKmInLKR;
    }

    
    

    public int getTypeID() {
        return TypeID;
    }

    public void setTypeID(int TypeID) {
        this.TypeID = TypeID;
    }

    public String getTypeName() {
        return TypeName;
    }

    public void setTypeName(String TypeName) {
        this.TypeName = TypeName;
    }

    public int getVehicleCapacity() {
        return vehicleCapacity;
    }

    public void setVehicleCapacity(int vehicleCapacity) {
        this.vehicleCapacity = vehicleCapacity;
    }

    public int getPricePerKmInLKR() {
        return pricePerKmInLKR;
    }

    public void setPricePerKmInLKR(int pricePerKmInLKR) {
        this.pricePerKmInLKR = pricePerKmInLKR;
    }
    
    
    
}

File: OnlineCabService-CustomerClient/src/main/webapp/vehicleType.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */




const url = "http://localhost:8080/OnlineCabService-Service/vehicletype";

function getVehicleType() {
    let id = document.getElementById("txtID").value;
    const options = {
        method : "GET"
    };
    
    fetch(url + id, options)
        .then(res => res.json())
        .then(data => {
            if (data !== null) {
               document.getElementById("").value = data.typeName;
               document.getElementById("").value = data.vehicleCapacity;
               document.getElementById("").value = data.PricePerKmInLKR;
            } else {
                alert("Not found");
            }

        });
}

function getAllVehicleTypes() {
    
}

function addVehicleType() {
    const vehicletype = {
        "typeID" : document.getElementById("").value,
        "typeName" : document.getElementById("").value,
        "vehicleCapacity" :  document.getElementById("").value,
        "pricePerKmInLKR" : document.getElementById("").value
    };
                
    const options = {
        method : "POST", 
        headers : {
            "content-type" : "application/json"
        },
        body : JSON.stringify(vehicletype)
    };
                
    fetch(url, options);
}

function updateVehicleType() {
    const vehicletype = {
        "typeID" : document.getElementById("").value,
        "typeName" : document.getElementById("").value,
        "vehicleCapacity" :  document.getElementById("").value,
        "pricePerKmInLKR" : document.getElementById("").value
    };
                
    const options = {
        method : "PUT", 
        headers : {
            "content-type" : "application/json"
        },
        body : JSON.stringify(vehicletype)
    };
                
    fetch(url, options);
}

function deleteVehicleType() {
    let id = document.getElementById("txtID").value;
    const options = {
        method : "DELETE"
    };
                
    fetch(url + id, options);
}

function clearVehicleType() {
    document.getElementById("").value = "";
    document.getElementById("").value = "";
    document.getElementById("").value = "";
    document.getElementById("").value = "";
    document.getElementById("").value = "";
}